### ISSUE TITLE
Trouble using extend_schema_field to output lists

### ISSUE THREAD

**Issue by w00kie** [tfranzel/drf-spectacular#1141]
tldr; I have a `SerializeMethodField` that returns a list of objects but `def-spectacular` does not auto-resolves its type and I can't find a way to hint it correctly using `@extend_schema_field`.

This is my view and I'm having trouble with the `issues` action:

```python
class TagViewSet(ModelViewSet):
    queryset = Tag.objects.all()
    serializer_class = TagSerializer

    @extend_schema(responses=EpicExtendedSerializer(many=True))
    @action(detail=True, serializer_class=EpicExtendedSerializer)
    def issues(self, request, pk=None):
        tag = self.get_object()
        issues = Epic.objects.filter(tags=tag)
        serializer = EpicExtendedSerializer(issues, many=True, context={"tag": tag})
        return Response(serializer.data)
```

And this is the Serializer used by the `issues` action:

```python
class EpicExtendedSerializer(serializers.ModelSerializer):
    stories = serializers.SerializerMethodField()
    tags = serializers.StringRelatedField(many=True)

    @extend_schema_field(StoryExtendedSerializer)
    def get_stories(self, epic):
        if self.context.get("tag"):
            stories = epic.stories.filter(tags=self.context.get("tag")).all()
        else:
            stories = epic.stories.all()
        serializers = StoryExtendedSerializer(stories, many=True, context=self.context)
        return serializers.data

    class Meta:
        model = Epic
        fields = "__all__"
```

Without the `@extend_schema_field(StoryExtendedSerializer)` statement, my schema outputs `"stories": "string"`. When I add it back, it will show the proper object format for stories but only as a single story.

I cannot find a way to either force `@extend_schema_field` to recognize it should be a `List[StoryExtendedSerializer]` or hint it towards auto-recognizing this.

**Comment by tfranzel** [comment]
Hi @w00kie,

so this looks totally fine to me. It is the default workflow to decorate the method and putting a serializer like `EpicExtendedSerializer` should always work.

Have you looked at the console output for a error message? Or run `manage.py spectacular --color --file foo.yml`. Looks to me like the serializer throws an error (due to some customization) and thus defaults to "string". 

Do you depend on `tag`? Might have to be graceful when its missing, since schema generation has no access to it.

**Comment by w00kie** [comment]
I generated the yaml file through the `manage.py spectacular` command as you suggested. There is no error message nor is there when fetching the schema from the server.

I get this for EpicExtended:

```yaml
EpicExtended:
  type: object
  properties:
    id:
      type: integer
      readOnly: true
    stories:
      allOf:
      - $ref: '#/components/schemas/StoryExtended'
      readOnly: true
```

When I would really expect something like this as I'm returning a list of stories:

```yaml
EpicExtended:
  type: object
  properties:
    id:
      type: integer
      readOnly: true
    stories:
      type: array
      items:
        $ref: '#/components/schemas/StoryExtended'
      readOnly: true
```

**Comment by tfranzel** [comment]
oops, I misread you question.

what about `@extend_schema_field(StoryExtendedSerializer(many=True))`? when you want a list, give it a `many` like you did with `EpicExtended` or am I still missing something.  It is all the same mechanics.

**Comment by w00kie** [comment]
YES! `@extend_schema_field(StoryExtendedSerializer(many=True))` is exactly what I needed. I just wasn't sure what was the correct syntax to signify that I had a list instead of a single object.

Thanks for pointing that out.

**Comment by tfranzel** [comment]
> Now that I see it, I just realized that I'm already using the correct syntax on the @extend_schema which I put on the view... 🤦

that was the source of my confusion, since you basically already knew what was needed.

this may seem a bit awkward to handle the list like that, but DRF is completely build around that idiom and the code flows very naturally due to us going the same route. anyways, cheers!
